LTTngCds/CookerData/LTTngContext.cs

```diff
@@ -31,6 +33,19 @@ internal LTTngContext(LTTngMetadataCustomization metadata, ICtfInputStream event
         /// </summary>
         public IReadOnlyDictionary<string, ICtfClockDescriptor> Clocks => this.metadata.Metadata.ClocksByName;
 
+        /// <summary>
+        /// Intern (de-duplicate) string to save memory.
+        /// </summary>
+        public string Intern(string input)

```

Comment:
There is already a Common.StringIntern

Reply:
That uses String.Intern(), which I don't want to use since it never releases the string reference while the process is running. I want to tie the string reference to the lifetime of the trace itself, so added a string pool to the LTTngcontext that'll go away when that trace is closed.

Reply:
True - I never liked that aspect of Common.StringIntern. Maybe we should just replace that implementation with this one?

Reply:
I briefly looked into that, and the StringIntern() needs to be on some type of Context object - which isn't always readily available in the places where it's currently used.  Good idea for a future improvement.